harness/determined/launch/autohorovod.py

```diff
@@ -90,28 +89,8 @@ def main(train_entrypoint: str) -> int:
 
     # Chief machine waits for every worker's sshd to be available.  All machines should be pretty
     # close to in-step by now because all machines just finished synchronizing rendezvous info.
-    deadline = time.time() + 20
-    for peer in info.container_addrs[1:]:
-        while True:
-            with socket.socket() as sock:
-                sock.settimeout(1)
-                try:
-                    # Connect to a socket to ensure sshd is listening.
-                    sock.connect((peer, DTRAIN_SSH_PORT))
-                    # The ssh protocol requires the server to serve an initial greeting.
-                    # Receive part of that greeting to know that sshd is accepting/responding.
-                    data = sock.recv(1)
-                    if not data:
-                        raise ValueError("no sshd greeting")
-                    # This peer is ready.
-                    break
-                except Exception:
-                    if time.time() > deadline:
-                        raise ValueError(
-                            f"Chief machine was unable to connect to sshd on peer machine at "
-                            f"{peer}:{DTRAIN_SSH_PORT}"
-                        )
-                    time.sleep(0.1)
+    for peer_addr in info.container_addrs[1:]:
+        util.check_sshd(peer_addr, time.time() + 5, DTRAIN_SSH_PORT)
```

Comment:
Let's not modify the deadline code.  20 seconds for all workers to become ready is not equivalent to 5 seconds for the first worker to come up, 5 more seconds for the next worker, etc.

Set a deadline once before the loop and pass it in repeatedly.